xwiki-rendering-macros/xwiki-rendering-macro-footnotes/src/main/java/org/xwiki/rendering/internal/macro/footnote/PutFootnotesMacro.java

```diff
@@ -228,6 +230,25 @@ public List<Block> execute(FootnoteMacroParameters parameters, String content, M
         return result;
     }
 
+    /**
+     * Repair a standalone footnote macro marker block by wrapping it in a paragraph if it is not already inline.
+     *
+     * @param macroMarkerBlock the macro marker block to repair
+     * @return the repaired macro marker block
+     */
+    private static MacroMarkerBlock repairStandaloneFootnote(MacroMarkerBlock macroMarkerBlock)
+    {
+        if (macroMarkerBlock.isInline()) {
+            return macroMarkerBlock;
+        } else {
+            // Wrap the macro marker block in a paragraph and make it inline.
+            MacroMarkerBlock result = new MacroMarkerBlock(macroMarkerBlock.getId(), macroMarkerBlock.getParameters(),
+                macroMarkerBlock.getContent(), macroMarkerBlock.getChildren(), true);
```

Comment:
@tmortagne I'm wondering if I should move the attributes of the block, too, any opinions on that?

Reply:
We don't have any attribute on MacroMarkerBlock right now AFAIK, but I feel it would be more future-proof to keep them on the new MacroMarkerBlock. We might have the same problem in other places where we do this kind of bloc refactoring.

That being said, shouldn't the content of a standalone footnote be standalone from the start ? It's not clear to me why it needs to be repaired.

Reply:
There are two things that make the footnote macro special:

1. The initially produced content doesn't stay in the footnote but is moved to the `putFootnotes` macro by the `putFootnotes` macro and new content is set by the `putFootnotes` macro. So initially, the `footnote` macro should always produce inline content as that's what fits better in `putFootnotes` (it puts it inside a list item).
2. While the new content that is added by the `putFootnotes` macro could be made standalone by wrapping it inside a paragraph, I felt that while technically correct, this won't provide a good user experience in the WYSIWYG editor. It doesn't make any sense for a footnote to be standalone and the WYSIWYG editor provides no way to change the footnote back to inline.

Phrased differently, what I would actually want is a way to indicate that the footnote macro doesn't support standalone mode. As this is not exactly common (I think), I did the quick solution to just wrap the footnote macro in a paragraph.